src/datepicker/day.js

```diff
@@ -87,9 +87,20 @@ export default class Day<T = Date> extends React.Component<
     const {range, value} = this.props;
     let date;
     if (Array.isArray(value) && range) {
-      if (!value.length || value.length > 1) {
-        date = [selectedDate];
-      } else if (this.dateHelpers.isAfter(selectedDate, value[0])) {
+      if ((!value[0] && !value[1]) || (value[0] && value[1])) {
```

Comment:
Can you please add some integration tests that validate the behavior here, and below?